homeassistant/components/switch/qwikswitch.py

```diff
@@ -4,18 +4,26 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/switch.qwikswitch/
 """
-import logging
+from homeassistant.components.qwikswitch import (
+    QSToggleEntity, DOMAIN as QWIKSWITCH)
+from homeassistant.components.switch import SwitchDevice
 
-DEPENDENCIES = ['qwikswitch']
+DEPENDENCIES = [QWIKSWITCH]
 
 
-# pylint: disable=unused-argument
-def setup_platform(hass, config, add_devices, discovery_info=None):
+async def async_setup_platform(hass, _, add_devices, discovery_info=None):
     """Add switches from the main Qwikswitch component."""
-    if discovery_info is None:
-        logging.getLogger(__name__).error(
-            "Configure Qwikswitch Switch component failed")
-        return False
+    qsusb = hass.data[QWIKSWITCH]
+    devs = [QSSwitch(id, qsusb) for id in discovery_info[QWIKSWITCH]]
```

Comment:
`id` is a list I think. Should each entity only get one id?

Reply:
Nevermind this comment. This is iterating that list.

Reply:
It is a single id as per above